packages/shared/src/netlifyCacheHandler.ts

```diff
@@ -0,0 +1,47 @@
+import { isDevelopmentFromPublishableKey } from './keys';
+
+/**
+ * Cache busting parameter for Netlify to prevent cached responses
+ * during handshake flows with Clerk development instances.
+ *
+ * Note: This query parameter will be removed in the "@clerk/clerk-js" package.
+ *
+ * @internal
+ */
+export const CLERK_NETLIFY_CACHE_BUST_PARAM = '__clerk_netlify_cache_bust';
+
+/**
+ * Prevents infinite redirects in Netlify's functions by adding a cache bust parameter
+ * to the original redirect URL. This ensures that Netlify doesn't serve a cached response
+ * during the handshake flow.
+ *
+ * The issue happens only on Clerk development instances running on Netlify. This is
+ * a workaround until we find a better solution.
+ *
+ * See {@link https://answers.netlify.com/t/cache-handling-recommendation-for-authentication-handshake-redirects/143969/1}
+ *
+ * @internal
+ */
+export function handleNetlifyCacheInDevInstance({
+  locationHeader,
+  requestStateHeaders,
+  publishableKey,
+}: {
+  locationHeader: string;
+  requestStateHeaders: Headers;
+  publishableKey: string;
+}) {
+  // eslint-disable-next-line turbo/no-undeclared-env-vars
+  const isOnNetlify = process.env.URL?.endsWith('netlify.app') || Boolean(process.env.NETLIFY_FUNCTIONS_TOKEN);
```

Comment:
`URL` and `NETLIFY_FUNCTIONS_TOKEN` are runtime environment variables injected by Netlify.

Reply:
Can't we use `process.env.NETLIFY`?

`URL` will not always end with `netlify.app` since it can also be your custom domain, as per https://docs.netlify.com/functions/environment-variables/#functions